yb-voyager/src/types/query_stats.go

```diff
@@ -27,3 +33,64 @@ type QueryStats struct {
 	MinExecTime     float64 `json:"min_exec_time" db:"min_exec_time"`
 	MaxExecTime     float64 `json:"max_exec_time" db:"max_exec_time"`
 }
+
+// MergeQueryStatsBasedOnQuery merges query stats with the same query text.
+// Used to aggregate stats from multiple source nodes (primary + replicas).
+// Mirrors the logic of pgss.MergePgStatStatementsBasedOnQuery but operates on QueryStats directly
+// to avoid conversion overhead between types and maintain a clean, direct code path.
+//
+// Merge key: Uses query text (not queryid) because:
+// - QueryIDs can differ for the same query across nodes due to user context, DB OIDs, or PG version differences
+// - Query text is the semantic truth - same text = same query, regardless of queryid
```

Comment:
let's document that ideally queryID should work, but for now we're using queryText, and that we will revisit this later.